packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts

```diff
@@ -52,74 +50,67 @@ type CSSModule = Module & {
-  for (const group of griffelChunk.groupsIterable) {
-    if (group.getModulePostOrderIndex(cssModule)) {
-      continue;
-    }
+function isGriffelCSSModule(module: Module): boolean {
+  if (isCSSModule(module)) {
+    if (Buffer.isBuffer(module.content)) {
+      const content = module.content.toString('utf-8');
 
-    // "mini-css-extract" requires an index to exist on modules. A module with an index will be filtered out and the plugin will throw
-    // https://github.com/webpack-contrib/mini-css-extract-plugin/blob/26334462e419026086856787d672b052cd916c62/src/index.js#L1133-L1140
-    group.setModulePostOrderIndex(
-      cssModule,
-      // It's bad to use private APIs, but it's more reliable than random indexes
-      // The same approach is used in Gatsby
-      // https://github.com/gatsbyjs/gatsby/blob/0b3c34c2bf932e5486ad2d0c3589bde6dc818661/packages/gatsby/src/utils/webpack/plugins/partial-hydration.ts#L455-L463
-      // https://github.com/webpack/webpack/blob/e184a03f2504f03b2e30091662df6630a99a5f72/lib/ChunkGroup.js#L98-L99
-      (group as ChunkGroup & { _modulePostOrderIndices: Map<Module, number> })._modulePostOrderIndices.size + 1,
-    );
+      return content.indexOf('/** @griffel:css-start') !== -1;
+    }
   }
+
+  return false;
 }
 
-function moveCSSModulesToGriffelChunk(compilation: Compilation) {
+function moveCSSModulesToGriffelChunk(compilation: Compilation, attachToMainEntryPoint: boolean) {
   const griffelChunk = compilation.namedChunks.get('griffel');
 
   if (!griffelChunk) {
     return;
   }
 
-  for (const chunk of compilation.chunks) {
-    if (chunk === griffelChunk) {
-      continue;
-    }
+  const matchingChunks = new Set<Chunk>();
+  let moduleIndex = 0;
 
-    // https://github.com/webpack-contrib/mini-css-extract-plugin/blob/26334462e419026086856787d672b052cd916c62/src/index.js#L693-L697
-    const cssModules = compilation.chunkGraph.getChunkModulesIterableBySourceType(chunk, 'css/mini-extract');
+  for (const module of compilation.modules) {
+    if (isGriffelCSSModule(module)) {
+      const moduleChunks = compilation.chunkGraph.getModuleChunksIterable(module);
 
-    if (typeof cssModules === 'undefined') {
-      continue;
-    }
+      for (const chunk of moduleChunks) {
+        compilation.chunkGraph.disconnectChunkAndModule(chunk, module);
+
+        for (const group of chunk.groupsIterable) {
+          group.setModulePostOrderIndex(module, moduleIndex++);
+        }
 
-    for (const cssModule of cssModules) {
-      if (!isGriffelCSSModule(cssModule as CSSModule)) {
-        continue;
+        matchingChunks.add(chunk);
       }
 
-      // https://github.com/webpack/webpack/blob/8241da7f1e75c5581ba535d127fa66aeb9eb2ac8/lib/Chunk.js#L245-L253
-      compilation.chunkGraph.disconnectChunkAndModule(chunk, cssModule);
-      compilation.chunkGraph.connectChunkAndModule(griffelChunk, cssModule);
+      compilation.chunkGraph.connectChunkAndModule(griffelChunk, module);
+    }
```

Comment:
The logic is almost the same: https://github.com/webpack/webpack/blob/b67626c7b4ffed8737d195b27c8cea1e68d58134/lib/optimize/SplitChunksPlugin.js#L1513